etc/kayobe/ansible/roles/opensearch/tasks/set_log_retention_policy.yml

```diff
@@ -0,0 +1,58 @@
+---
+- name: Check required variables are set
+  fail:
+    msg: "Please set the variable {{ item.key }} in opensearch.yml. For example: {{ item.key }}: {{ item.value }} "
+  when: "{{ item.key }} is not defined"
+  with_dict:
+    - "opensearch_internal_endpoint": '"http://{{ kolla_internal_fqdn }}:9200"'
+    - "opensearch_log_index_pattern": '"flog-*"'
+
+- name: Wait for OpenSearch to become ready
+  become: true
+  uri:
+    url: "{{ opensearch_internal_endpoint }}/_cluster/stats"
+    status_code: 200
+  register: result
+  until: result.get('status') == 200
+  retries: 30
+  delay: 2
+  run_once: true
```

Comment:
This should probably include changed_when: false, since it will never change anything but always runs.

Reply:
I agree in general. I think in this case it's handled internally by the module. Eg. if the service is running, it looks like this:

![image](https://github.com/stackhpc/stackhpc-kayobe-config/assets/1140727/2536f475-c9b8-46a2-82da-c8725d0db141)

If the service is down for a bit:

![image](https://github.com/stackhpc/stackhpc-kayobe-config/assets/1140727/6c0c7fd7-3ed3-465a-a466-9f567e8ac255)